### PR TITLE
Smart servo ros1 carousel code

### LINKED ISSUE #578: Carousel and funnel control for ATP test payload
Similarly to the old now defunct science payload, we will need a way to turn a carousel containing vials. This would be in a part of the science module just like how the Teensy code for the wheel and arm are. 

There are limit swtiches on board in order to detect what position the carousel is in. The vials are to be positioned under a funnel for collecting soil samples. 

There is likely some old work for the old science payload in a branch that can be reused.

### PR DESCRIPTION
**Pull request by vexandmore**
# Assignee Section

## Description

- Added teensy code to control the smart servo based carousel
- Made CommsNode work in local mode
- Add science commands to comms node

### Steps for Testing
These steps assume a test not on rover
1. Start roscode
2. run the CommsNode.py node with 'local' as an argument ie `rosrun mcu_command CommsNode.py local`
4. Plug in the teensy to usb port, with the smart servo on serial 5. Note a level shifter is needed.
5. Send over commands, eg `rostopic pub /science_command std_msgs/String "data: 'next_test_tube'"`

closes #578

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

For Pull Requests that do not include code changes, it is not required to perform the tests above.

### CODE REVIEW

File: robot/rospackages/src/mcu_control/scripts/CommandParser.py
Comment (RealWilliamWells) [inline robot/rospackages/src/mcu_control/scripts/CommandParser.py:33]: This is a very smart solution, good stuff

**Review by RealWilliamWells — APPROVED** [review]
Excellent work! All of your code is very clean and easy to follow

File: robot/rospackages/src/mcu_control/scripts/CommandParser.py
Comment (vexandmore) [reply robot/rospackages/src/mcu_control/scripts/CommandParser.py:33]: yeah I had seen in a talk a mention of ruby's method missing, found out python has a similar thing

**Review by RealWilliamWells — APPROVED** [review]
Everything should be good now

**Comment by RealWilliamWells** [comment]
@vexandmore Can you review my changes and then approve? Once you approve, I will force merge this into master

**Pull request merged**
